numpy/core/fromnumeric.py

```diff
@@ -446,13 +508,13 @@ def put(a, ind, v, mode='raise'):
     array([ 0,  1,  2,  3, -5])
 
     """
-    try:
-        put = a.put
-    except AttributeError:
+    # Because the putting is done in-place, we need to enforce
+    # that the object being passed in is an instance of ndarray
+    if not isinstance(a, mu.ndarray):
```

Comment:
Is there really no way to currently call `np.put` on a non-ndarray and get a correct results? If not, this is a breaking API change that we need to be extreme cautious about.